### PR TITLE
*: move `StmtCtx.ErrAutoincReadFailedAsWarning` to errctx

### LINKED ISSUE #49991: use `errctx.Context` to handle error `ErrAutoincReadFailed`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #49991

### What changed and how does it work?

Use `errctx.Context` to handle `ErrAutoincReadFailed` to make error handling decouple with `stmtctx`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/sessionctx/stmtctx/stmtctx.go [hunk 1]
```diff
@@ -443,17 +443,17 @@ func NewStmtCtxWithTimeZone(tz *time.Location) *StatementContext {
...
 // Reset resets a statement context
 func (sc *StatementContext) Reset() {
 	*sc = StatementContext{
-		ctxID:   stmtCtxIDGenerator.Add(1),
-		typeCtx: types.NewContext(types.DefaultStmtFlags, time.UTC, sc),
+		ctxID: stmtCtxIDGenerator.Add(1),
 	}
-	sc.initErrCtx()
+	sc.typeCtx = types.NewContext(types.DefaultStmtFlags, time.UTC, sc)
+	sc.errCtx = newErrCtx(sc.typeCtx, errctx.LevelMap{}, sc)
```
Comment (YangKeao) [inline pkg/sessionctx/stmtctx/stmtctx.go:456]: Thanks for this modification. It shows the dependency of `errCtx` and it's much clearer.

**Review by YangKeao — APPROVED** [review]
LGTM :+1:

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**